util/nats-server-hardened.service

```diff
@@ -0,0 +1,43 @@
+[Unit]
+Description=NATS Server
+After=network.target ntp.service
+
+[Service]
+Type=simple
+ExecStart=/usr/sbin/nats-server -c /etc/nats-server.conf
+ExecReload=/bin/kill -s HUP $MAINPID
+ExecStop=/bin/kill -s SIGINT $MAINPID
+User=nats
+Group=nats
+
+# Hardening
+CapabilityBoundingSet=
+LockPersonality=true
+MemoryDenyWriteExecute=true
+PrivateDevices=true
+PrivateTmp=true
+PrivateUsers=true
+PrivateIPC=true
+ProcSubset=pid
+ProtectClock=true
+ProtectControlGroups=true
+ProtectHome=true
+ProtectHostname=true
+ProtectKernelLogs=true
+ProtectKernelModules=true
+ProtectKernelTunables=true
+ProtectProc=invisible
```

Comment:
This one I'm not comfortable with, unless added to a section which calls out that it's for very new systemd?  This isn't in `systemd 245` as supplied by Ubuntu 20.04 LTS.

NixOS being able to benefit from shipping the latest and greatest is good, but this PR would be even better if we could call out those which won't work in the latest LTS from Ubuntu, as that will reduce the number of times we're asked for support by people who see "hardened" and try it, without understanding or the ability to trade off.

Reply:
Thanks for your feedback!

I left out `NoNewPrivileges=true` because it's implied by other flags. But now that you mention it, having it in the file explicitly makes sense.

I've added the `LimitNOFILE` to the set. Also mentioned the resource control options. I won't put the resource control options in the NixOS NATS module configuration, as the weight values depend on the user's use of the module (usage, system load, capacity, etc.). In NixOS it's fairly easy for admins to extend the NATS systemd service: `systemd.services.nats.serviceConfig.CPUWeight = 1000;`

`ProtectProc` and `PrivateIPC` have a comment now with the minimum required systemd version. I did some testing, and it appears that systemd happily ignores options that it doesn't know yet. Even my unstable NixOS only has systemd `247.6`, which doesn't support `PrivateIPC`, yet I don't see any warning in `systemctl status nats` or `journalctl -u nats`.

Reply:
I searched in the documentation on what systemd does when it encounters an unknown option. For the `[Unit]` section the [docs](https://www.freedesktop.org/software/systemd/man/systemd.unit.html) say:

> Unit files may contain additional options on top of those listed here. If systemd encounters an unknown option, it will write a warning log message but continue loading the unit. If an option or section name is prefixed with X-, it is ignored completely by systemd. Options within an ignored section do not need the prefix. Applications may use this to include additional information in the unit files.

However, the [docs for the `[Service]` section](https://www.freedesktop.org/software/systemd/man/systemd.service.html)  doesn't mention how unknown options are dealt with.

So, a comment makes sense. Then people know what to expect.

Reply:
I've commented out the resource control options because I'm afraid that some users will copy it without reading the comments, and thus have their NATS running at 1% CPU limit.

Reply:
Can this PR be merged @philpennock?

Reply:
I am very sorry, you deserved a more timely response than this. Poor GitHub notifications management on my part.

Thank you for the contribution and for addressing my feedback, I'm very happy with the result and have given my GH 👍 approval.